### PR TITLE
[Android] WiFi stats for mobile

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

**Comment by damencho** [comment]
All the other parts are merged, callstats is updated, so this is ready to be merged if approved.

File: android/sdk/src/main/java/org/jitsi/meet/sdk/WiFiStatsModule.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+                    WifiInfo wifiInfo = wifiManager.getConnectionInfo();
+
+                    if (wifiInfo.getNetworkId() == -1) {
+                        promise.reject(new Exception("Wifi not connected"));
+                        return;
+                    }
+
+                    int rssi = wifiInfo.getRssi();
+                    int signalLevel = WifiManager.calculateSignalLevel(
```
Comment (paweldomas) [inline android/sdk/src/main/java/org/jitsi/meet/sdk/WiFiStatsModule.java]: Wouldn't this look better (similar in few other places) ?
```
int signalLevel
    = WifiManager.calculateSignalLevel(
            rssi, SIGNAL_LEVEL_SCALE);
```

Comment (damencho) [reply android/sdk/src/main/java/org/jitsi/meet/sdk/WiFiStatsModule.java]: Ok, I can add those.

**Comment by paweldomas** [comment]
@saghul how do you know it supports only ipv4 ? I don't see that anywhere in the docs ?

**Comment by saghul** [comment]
@paweldomas I don't recall where I saw it (maybe stackoverflow?!) but since you cannot represent an IPv6 address as an `int`, makes sense, right?

**Comment by paweldomas** [comment]
No, actually it doesn't make sense. I'm not sure how do you want to represent even IPv4 with an int. It should return https://developer.android.com/reference/java/net/InetAddress.html or am I looking at completely different thing ?

https://developer.android.com/reference/java/net/NetworkInterface.html#getInetAddresses()

**Comment by saghul** [comment]
@paweldomas Look at https://developer.android.com/reference/android/net/wifi/WifiInfo.html#getIpAddress()

It returns a binary packed IPv4 address, as a 32 bit integer.

**Comment by paweldomas** [comment]
Oh that's different thing than I was looking at.

So I won't be able to test that easily or at least it will be as hard as for you or for Damencho, because the IPv6 network that I get is not a wifi network, but a cellular one.

**Comment by saghul** [comment]
Ah, I see. I'll try to test it tomorrow, since I use a cable here.

**Comment by saghul** [comment]
Confirmed, the patch works fine. On an IPv6-only network with NAT64 + DNS64, no IP address is detected, since the WiFiInfo module does not have IPv6 capability.

**Pull request merged**